tests/integration/test_charm.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# Copyright 2024 user
+# See LICENSE file for licensing details.
+
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./charmcraft.yaml").read_text())
+APP_NAME = METADATA["name"]
+TEST_ACTION_NAME = "test"
+# Allow e2e tests to run with up to 2 non-ready nodes
+EXTRA_ARGS = "-allowed-not-ready-nodes 2"
+# Run reduced test suite
+SKIP_TESTS = r"\[(Flaky|Slow|Conformance|Feature:.*)\]"
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    # Build and deploy charm from local source folder
+    charm = await ops_test.build_charm(".")
+
+    await ops_test.model.deploy(charm, application_name=APP_NAME, series="focal")
```

Comment:
let the juju model-defaults figure that out.  focal will be eol in a while

```suggestion
    await ops_test.model.deploy(charm, application_name=APP_NAME)
```